cmd/run.go

```diff
@@ -442,3 +451,53 @@ func handleSummaryResult(fs afero.Fs, stdOut, stdErr io.Writer, result map[strin
 
 	return consolidateErrorMessage(errs, "Could not save some summary information:")
 }
+
+// StartOutputs spins up all configured outputs, giving the thresholds to any
+// that can accept them. And if some output fails, stop the already started
+// ones. This may take some time, since some outputs make initial network
+// requests to set up whatever remote services are going to listen to them.
+func StartOutputs(
```

Comment:
I literally otherwise needed to thread too much stuff through the engine, and decided that moving it out will be much easier.

THis probably should be method on some object having `outputs` and `logger` as fields ... but I didn't want to make additional changes (this will be a theme in this comments and the whole WIP PR :P )

Reply:
I will drop this and make a separate PR if we want to move them out? 
WDYT @na-- @imiric @codebien